plugins/depthCamera/src/DepthCameraDriver.cpp

```diff
@@ -120,23 +127,25 @@ bool GazeboYarpDepthCameraDriver::close()
     // Copy & paste like a Monkey
     if (this->m_updateImageFrame_Connection.get())
     {
-        m_imageCameraPtr->DisconnectNewImageFrame(this->m_updateImageFrame_Connection);
+        m_depthCameraPtr->DisconnectNewImageFrame(this->m_updateImageFrame_Connection);
         this->m_updateImageFrame_Connection = gazebo::event::ConnectionPtr();
     }
 
     if (this->m_updateRGBPointCloud_Connection.get())
     {
-        m_imageCameraPtr->DisconnectNewImageFrame(this->m_updateRGBPointCloud_Connection);
+        m_depthCameraPtr->DisconnectNewImageFrame(this->m_updateRGBPointCloud_Connection);
```

Comment:
Related issue:
* https://bitbucket.org/osrf/gazebo/issues/1865/rendering-depthcamera-does-not-output
* http://answers.gazebosim.org/question/5469/does-gazebo_ros_depth_camera-with-points-output-produce-a-color-point-cloud/